apps/ejabberd/src/event_pusher/mod_event_pusher_http_notification.erl

```diff
@@ -0,0 +1,92 @@
+%%%----------------------------------------------------------------------
+%%% File    : mod_http_notification.erl
+%%% Author  : Baibossynv Valery <baibossynov.valery@gmail.com>
+%%% Purpose : Message passing via http
+%%% Created : 16 Dec 2015 by Baibossynv Valery <baibossynov.valery@gmail.com>
+%%%----------------------------------------------------------------------
+
+-module(mod_event_pusher_http_notification).
+-author("baibossynov.valery@gmail.com").
+
+-behaviour(gen_mod).
+-behaviour(mod_event_pusher).
+
+-include("mod_event_pusher_events.hrl").
+-include_lib("ejabberd/include/jlib.hrl").
+
+-callback should_make_req(Packet :: exml:element(), From :: jid(), To :: jid()) -> boolean().
+
+%% API
+-export([start/2, stop/1, push_event/2]).
+
+-include("ejabberd.hrl").
+-include("jlib.hrl").
+
+-define(DEFAULT_POOL_NAME, http_pool).
+-define(DEFAULT_PATH, "").
+
+-define(SENT_METRIC, [mod_http_notifications, sent]).
+-define(FAILED_METRIC, [mod_http_notifications, failed]).
+-define(RESPONSE_METRIC, [mod_http_notifications, response_time]).
+
+start(Host, _Opts) ->
+    ensure_metrics(Host),
+    ok.
+
+stop(_Host) ->
+    ok.
+
+push_event(_, #chat_event{direction = in, from = From, to = To, packet = Packet}) ->
+    Body = exml_query:path(Packet, [{element, <<"body">>}, cdata], <<>>),
+    Mod = get_callback_module(),
+    case Mod:should_make_req(Packet, From, To) of
```

Comment:
For some reason Elvis expects `-callback` after `-export` and expects `-export` after `-include`s.